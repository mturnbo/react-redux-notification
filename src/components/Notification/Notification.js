import React from 'react';
import PropTypes from 'prop-types';
import './Notification.scss';

const Notification = props => {
  let notificationClass = 'notification';
  notificationClass += props.display ? ' show' : '';
  return (
    <div className={notificationClass}>
      <button type="button" className="notification-close" onClick={props.closeHandler}>X</button>
      <div className="notification-title">{props.title}</div>
      <div className="notification-message">{props.message}</div>
    </div>
  );
};

Notification.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  display: PropTypes.bool.isRequired,
  closeHandler: PropTypes.func.isRequired
};

export default Notification;
