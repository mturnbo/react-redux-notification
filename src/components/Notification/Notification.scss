.notification {
  position: fixed;
  top: 80px;
  right: 5px;
  padding: 10px;
  background-color: orange;
  color: #FFFFFF;
  border: 1px solid #666666;
  border-radius: 5px;
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.2);
  transition: 0.3s all;
  opacity: 0;
  transform: translate(0, 100px);
  min-width: 250px;
  max-width: 500px;
  text-align: left;

  &.show {
    opacity: 1;
    transform: translate(0, 0);
  }
}

.notification-close {
  border: 1px solid #000000;
  color: #000000;
  margin-right: 0 !important;
  margin-left: auto !important;
  display: block !important;
}

.notification-title {
  font-weight: bold;
  color: #333333;
  border-bottom: 1px solid #000000;
  margin-bottom: 5px;
}

.notification-message {
  color: #000000;
}
