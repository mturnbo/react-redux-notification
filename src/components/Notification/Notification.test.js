import React from 'react';
import { mount } from 'enzyme';
import Notification from './index';

const mockHandler = jest.fn();

const wrapper = mount(<Notification title="Test" message="Blah blah blah" display={true} closeHandler={mockHandler} />);

test('Notification Component Renders', () => {
  expect(wrapper).toMatchSnapshot();
});
