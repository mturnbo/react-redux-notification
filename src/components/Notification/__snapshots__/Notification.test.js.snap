// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Component Renders 1`] = `
<Notification
  closeHandler={[MockFunction]}
  display={true}
  message="Blah blah blah"
  title="Test"
>
  <div
    className="notification show"
  >
    <button
      className="notification-close"
      onClick={[MockFunction]}
      type="button"
    >
      X
    </button>
    <div
      className="notification-title"
    >
      Test
    </div>
    <div
      className="notification-message"
    >
      Blah blah blah
    </div>
  </div>
</Notification>
`;
